goosebit/settings/schema.py

```diff
@@ -61,9 +63,17 @@ def settings_customise_sources(
         dotenv_settings: PydanticBaseSettingsSource,
         file_secret_settings: PydanticBaseSettingsSource,
     ) -> tuple[PydanticBaseSettingsSource, ...]:
+        # skip arg parsing if running pytest
+        if "pytest" not in sys.modules:
+            cli_args = parser.parse_args()
+            return (
+                init_settings,
+                YamlConfigSettingsSource(settings_cls, cli_args.settings_file),
+                env_settings,
+                file_secret_settings,
+            )
```

Comment:
Not a fan of this personally, but I cant think of a better option with the current project structure...

Reply:
This looks awful.